src/lib.rs

```diff
@@ -893,44 +850,18 @@ impl IngestProcessor {
-                    match &self.opts.starting_offsets {
-                        StartingOffsets::Earliest => {
-                            info!(
-                                "Seeking consumer to earliest offsets for partition {} of topic {}",
-                                p, self.topic
-                            );
-                            self.consumer.seek(
-                                &self.topic,
-                                *p,
-                                Offset::Beginning,
-                                Timeout::Never,
-                            )?;
-                        }
-                        StartingOffsets::Latest => {
-                            info!(
-                                "Seeking consumer to latest offsets for partition {} of topic {}",
-                                p, self.topic
-                            );
-                            self.consumer
-                                .seek(&self.topic, *p, Offset::End, Timeout::Never)?;
-                        }
-                        StartingOffsets::Explicit(starting_offsets) => {
-                            if let Some(offset) = starting_offsets.get(p) {
-                                info!("Seeking consumer to offset {} for partition {} of topic {}. No offset is stored in delta log but explicit starting offsets are specified.", offset, p, self.topic);
-                                self.consumer.seek(
-                                    &self.topic,
-                                    *p,
-                                    Offset::Offset(*offset),
-                                    Timeout::Never,
-                                )?;
-                            } else {
-                                warn!("Not seeking consumer. Offsets are explicit, but an entry is not provided for partition {} of topic {}. `auto.offset.reset` from additional kafka settings will be used.", p, self.topic);
-                                // TODO: may need to lookup auto.offset.reset and force seek instead
-                            }
-                        }
-                    };
-                }
+                None => match self.opts.auto_offset_reset {
+                    AutoOffsetReset::Earliest => {
+                        info!("Seeking consumer to beginning for partition {} of topic {}. Partition has no stored offset but 'auto.offset.reset' is earliest", p, self.topic);
+                        self.consumer
+                            .seek(&self.topic, *p, Offset::Beginning, Timeout::Never)?;
+                    }
+                    AutoOffsetReset::Latest => {
+                        info!("Seeking consumer to end for partition {} of topic {}. Partition has no stored offset but 'auto.offset.reset' is latest", p, self.topic);
+                        self.consumer
+                            .seek(&self.topic, *p, Offset::End, Timeout::Never)?;
+                    }
+                },
```

Comment:
Given https://github.com/delta-io/kafka-delta-ingest/pull/92/files#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R1193-R1197 - I don't think this manual seek is necessary.

Reply:
well, just to make it obvious but yeah, it'll end up relatively the same

Reply:
There is a bit of a functional difference as well. When we set earliest/latest in `auto.offset.reset` Kafka config, the first offset we receive will be appropriate. By doing a manual seek here, we are burning some cpu and network unnecessarily.

Reply:
yes, but in both cases we'll end up with the latest offsets, e.g. just with a some messages off, which doesnt matter in term of latestm anyway, gonna update a suggested